src/main/java/com/novemberain/quartz/mongodb/dao/CalendarDao.java

```diff
@@ -48,14 +49,16 @@ public boolean remove(String name) {
         return false;
     }
 
-    public Calendar retrieveCalendar(String calName) {
+    public Calendar retrieveCalendar(String calName) throws JobPersistenceException {
         if (calName != null) {
-            // TODO
-            throw new UnsupportedOperationException();
+            Bson searchObj = Filters.eq(CALENDAR_NAME, calName);
+            Document doc = calendarCollection.find(searchObj).first();
+            Binary serializedCalendar = doc.get(CALENDAR_SERIALIZED_OBJECT, Binary.class);
```

Comment:
This line may produce NPE when the calendar will not be found, so check is needed here.